src/odemis/driver/test/symphotime_test.py

```diff
@@ -0,0 +1,142 @@
+'''
+
+CONFIG_SCANNER = {
+    "name": "Test scanner",
+    "role": "scanner",
+}
+
+# Config to use with the symphotime simulator
+CONFIG_SYMPHOTIME_SIM = {
+    "name": "Test Symphotime Controller",
+    "role": "detector",
+    "children": { "scanner": CONFIG_SCANNER },
+    "host": "localhost",
+    }
+
+MD = {
+    model.MD_DESCRIPTION: "Measurement description",
+    model.MD_PIXEL_SIZE: (5e-6, 5e-6),
+    model.MD_DWELL_TIME: 5e-6
+    }
+
+class TestSymphotime(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        # note: The symphotime simulator must be running before we start the tests. 
+        self.controller = symphotime.Controller(**CONFIG_SYMPHOTIME_SIM)
+        self.controller.updateMetadata(MD)
+
+    @classmethod
+    def tearDownClass(self):
+        self.controller.terminate()
+    
+    def test_acquisition(self):
+        # Test the starting and stopping of acquisition. 
+        done = threading.Event()
+        
+        for filename in ["file1.ptu", "big$file.ptu"]:
+            for directory in ["MyMeasurement", "TestGroup"]:
+                done.clear()
+                self.controller.scanner.filename.value = filename
+                self.controller.scanner.directory.value = directory
+
+                # This function will execute once the measurement completes.
+                def callback(flow, array):
+                    self.assertEqual(array, [[0]])
+                    self.assertEqual(self.controller.scanner.filename.value, filename)
+                    self.assertEqual(self.controller.scanner.directory.value, directory)
+                    self.assertEqual(array.metadata[model.MD_DESCRIPTION], MD[model.MD_DESCRIPTION])
+                    self.assertEqual(array.metadata[model.MD_PIXEL_SIZE], MD[model.MD_PIXEL_SIZE])
+                    self.assertEqual(self.controller.scanner.dwellTime.value, array.metadata[model.MD_DWELL_TIME])
+                    # unsub
+                    self.controller.data.unsubscribe(callback)
+                    self.assertFalse(self.controller.isMeasuring())
+                    done.set()
+
+                # Start measuring
+                self.controller.data.subscribe(callback)
+
+                # Should start measuring now
+                self.assertTrue(self.controller.isMeasuring())
+
+                # Wait until the measurement completes
+                done.wait(30)
+                self.assertFalse(self.controller.isMeasuring())
+
+    def test_bad_filename(self):
+        # Test if the filename will get an extension added.
+        filename = "filename"
+        self.controller.scanner.filename.value = filename
+        self.assertEqual(self.controller.scanner.filename.value, filename + '.ptu')
+
+        # Test that a filename that is to long is rejected.
+        with self.assertRaises(ValueError):
+            self.controller.scanner.filename.value = 'a' * 256
+            self.controller.scanner.filename.value = "filename.ome.tiff"
```

Comment:
With `.ome.tiff`, it ought to work. That's actually what the Odemis GUI will set. => Check that the (final) filename is something that SymPhoTime is happy with.
Anyway, for now this line is not even executed, as the previous line raises an error.